src/selection.ts

```diff
@@ -200,6 +198,25 @@ export interface IntervalSelectionConfig extends BaseSelectionConfig<'interval'>
   mark?: BrushConfig;
 }
 
+export interface RegionSelectionConfig extends BaseSelectionConfig<'region'> {
+  /**
+   * A region selection also adds a path mark to depict the
+   * shape of the region. The `mark` property can be used to customize the
+   * appearance of the mark.
+   *
+   * __See also:__ [`mark` examples](https://vega.github.io/vega-lite/docs/selection.html#mark) in the documentation.
+   */
+  mark?: BrushConfig;
+
+  /**
+   * An array of field names whose values must match for a data tuple to
+   * fall within the selection.
+   *
+   * __See also:__ The [projection with `encodings` and `fields` section](https://vega.github.io/vega-lite/docs/selection.html#project) in the documentation.
+   */
+  fields?: FieldName[];
```

Comment:
How should this property work?  Shouldn't the fields always match the x and y encodings? 

It would be good to add an example for this property if it should be supported, or otherwise we should remove them.

Reply:
Hi, looking at the vega counterpart for this PR
https://github.com/vega/vega/pull/3388/files#diff-547caeeef0ceae19c314c72edcd93ab9aac4a9462ada8fbae25ec7d37fad147b

the intersection test will always use x and y coordinates, if this is what you meant.
I think I took the interval selection config as a starting point, can I just remove the fields property if it is not needed?

Reply:
> Shouldn´t the fields property be always [SELECTION_ID] like in the default config for the point selection?

Point selection indeed uses `[SELECTION_ID]` by default. However, one can configure the `fields` to be different for point selection, thus having the `fields` property for users to customize makes sense.  

So the question is whether customizing this for lasso ever make sense.  
If it should always be a constant value and can't be customized, we shouldn't expose it in the specification. 

If it can be customized, better add an example to demonstrate at least one practical use case.

Reply:
@dvmoritzschoefl: the typings here generate the JSON Schema and/or serve as the public interface for what Vega-Lite users believe can be specified. You're quite right that the lasso selection's predicate uses `SELECTION_ID`, but it does not make sense to customize it to use other fields (i.e., your predicate function is hard coded to `vlSelectionIdTest`) or for customizing the field to affect how lassoing works operationally/mechanically (i.e., how we can get a uni-dimensional brush by specifying `"encodings": ["x"]` for interval selections).

So I think @kanitw is quite right that this property should be dropped here — great catch @kanitw!

Reply:
Ok, thanks for the clarification that makes sense. So removing both this property and the initial config for it should do the trick?

Reply:
@kanitw It seems that the CLI is generating the faulty images after a new commit. Is there an easy way to test why it is generating empty images? I tested the examples locally and they seem correct. Also the other images are correctly created in the runtime tests.

Reply:
@kanitw @arvind Was I correct under the assumption that the 2 images were in the wrong folder? Is it resolved then or is there another thing to do before this can be merged?